core/src/main/java/org/apache/calcite/util/RangeSets.java

```diff
@@ -105,6 +110,45 @@ private RangeSets() {}
     return 0;
   }
 
+  /**
+   * Estimates the number of distinct values for a range (if possible).
+   * A null is returned if the number of distinct values is infinity or unknown.
+   */
+  public static <C extends Comparable<C>> @Nullable Double numDistinctVals(
+      Range<C> range, RelDataType type) {
+    if (RangeSets.isPoint(range)) {
+      return 1.0;
```

Comment:
hmmm Why 1.0 when range is a point (i.e. not return null)?

Reply:
Returning null means we do not have an estimation, or equivalently, the number of distinct values is infinity.

When the range is a point (e.g. `x = 1`), we estimate that there is only one distinct value, which is likely to be true in practice.